pkg/imagestream/imagestream.go

```diff
@@ -267,21 +267,37 @@ func (is *imageStream) Exists(ctx context.Context) (bool, *rerrors.Error) {
 	return true, nil
 }
 
-func (is *imageStream) localRegistry(ctx context.Context) (string, *rerrors.Error) {
+func (is *imageStream) localRegistry(ctx context.Context) ([]string, *rerrors.Error) {
 	stream, rErr := is.imageStreamGetter.get()
 	if rErr != nil {
-		return "", convertImageStreamGetterError(rErr, fmt.Sprintf("localRegistry: failed to get image stream %s", is.Reference()))
+		return nil, convertImageStreamGetterError(rErr, fmt.Sprintf("localRegistry: failed to get image stream %s", is.Reference()))
 	}
 
+	var localNames []string
+
 	local, err := imageapi.ParseDockerImageReference(stream.Status.DockerImageRepository)
 	if err != nil {
-		return "", rerrors.NewError(
+		return nil, rerrors.NewError(
+			ErrImageStreamUnknownErrorCode,
+			fmt.Sprintf("localRegistry: unable to parse dockerImageRepository %q", stream.Status.DockerImageRepository),
+			err,
+		)
+	} else if len(local.Registry) != 0 {
+		localNames = append(localNames, local.Registry)
+	}
+
+	public, err := imageapi.ParseDockerImageReference(stream.Status.PublicDockerImageRepository)
+	if err != nil {
+		return nil, rerrors.NewError(
```

Comment:
i think you should still return localNames here.  The caller of localRegistry() isn't even checking the error (which should perhaps be fixed) and we at least want to filter the internal hostname, even if we can't parse the external one